api/src/services/DatastreamManager.ts

```diff
@@ -18,13 +18,27 @@ class DatastreamManager {
 
     async uploadFile(pid: string, stream: string, filepath: string, mimeType: string): Promise<void> {
         const fedoraObject = FedoraObject.build(pid);
+
         if (this.hasValidMimeType(stream, mimeType)) {
-            await fedoraObject.updateDatastreamFromFile(filepath, stream, mimeType);
+            try {
+                await fedoraObject.updateDatastreamFromFile(filepath, stream, mimeType);
+            } catch (error) {
+                console.log(error);
+                if ((error?.message || "").includes("410")) {
```

Comment:
I think this is a reasonably safe test, but I wonder if we could make it more explicit by using a more specific exception class for HTTP errors and storing the HTTP code in a property on the thrown exception object.

Reply:
Made change at the Fedora.ts level.